cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -66,24 +64,22 @@ func newDevicePlugin(sysfsDir, devfsDir string, sharedDevNum int) *devicePlugin
 
 func (dp *devicePlugin) Scan(notifier dpapi.Notifier) error {
 	for {
-		devTree, err := dp.scan()
-		if err != nil {
-			return err
```

Comment:
we could also keep returning `error` from `scan()` to keep the unit tests as is and trap the error here without returning?

Reply:
ah, like that idea @mythi Didn't occur to me as I started by fixing the lower func, and the changes rippled up... have rewritten the patch - it is now a one line change!
Pushed.